js/app/modules/selection/search.js

```diff
@@ -10,20 +11,19 @@ const Search = new Module.Class({
     Name: 'Selection.Search',
     Extends: Xapian.Xapian,
 
-    // Selection.Xapian implementation
-    on_history_changed: function (history) {
-        let item = history.get_current_item();
-        if (item.query) {
-            this._item = item;
+    _init: function (props) {
+        this.parent(props);
+        HistoryStore.get_default().connect('notify::current-query', () => {
             this._set_needs_refresh(true);
-        }
+        });
     },
 
     construct_query_object: function (limit, query_index) {
-        if (query_index > 0)
+        let query = HistoryStore.get_default().current_query;
+        if (query_index > 0 || query.length === 0)
```

Comment:
do we want to be returning `null` if query is empty string? Does `engine` handle `null` queries nicely?